src/server/ua_server_worker.c

```diff
@@ -563,6 +569,50 @@ UA_StatusCode UA_Server_run_startup(UA_Server *server) {
         result |= nl->start(nl, server->config.logger);
     }
 
+
+#ifdef UA_ENABLE_DISCOVERY_MULTICAST
+    if (server->config.applicationDescription.applicationType == UA_APPLICATIONTYPE_DISCOVERYSERVER) {
+
+        char *appName = (char *)malloc(server->config.mdnsServerName.length +1);
+        memcpy(appName, server->config.mdnsServerName.data, server->config.mdnsServerName.length);
+        appName[server->config.mdnsServerName.length] = '\0';
+
+        for(size_t i = 0; i < server->config.networkLayersSize; i++) {
+            UA_ServerNetworkLayer* nl = &server->config.networkLayers[i];
+            UA_UInt16 port = 0;
+            char hostname[256]; hostname[0] = '\0';
+            const char *path;
+            {
+                char* uri = (char *)malloc(sizeof(char) * nl->discoveryUrl.length + 1);
+                strncpy(uri, (char*) nl->discoveryUrl.data, nl->discoveryUrl.length);
+                uri[nl->discoveryUrl.length] = '\0';
+                UA_StatusCode retval;
+                if ((retval = UA_EndpointUrl_split(uri, hostname, &port, &path)) != UA_STATUSCODE_GOOD) {
+                    if (retval == UA_STATUSCODE_BADOUTOFRANGE)
+                        UA_LOG_WARNING(server->config.logger, UA_LOGCATEGORY_NETWORK, "Server url '%s' size invalid", uri);
```

Comment:
This return value may also mean that the URL is invalid, e.g. port too long, not only number of chars